Sources/Routing/Protocols/NavigationPathManageable.swift

```diff
@@ -28,10 +29,19 @@ extension NavigationPathManageable {
         }
         path.removeLast(count)
     }
+    
+    // Function to pop the navigation stack up to a specified destination
+    public func pop(to destination: Destination) {
+         // Check if the destination exists in the stack
+          if let index = path.lastIndex(where: { $0 == destination }) {
+              // Remove destinations above the specified destination
+              path = Array(path.prefix(upTo: index + 1))
```

Comment:
We could make this more readable if we had an array extension..

Maybe something like this?
```swift
extension Array {
    /// Truncates the array up to the specified index.
    ///
    /// This method mutates the original array, keeping the elements up to the given index.
    /// If the index is greater than the array's count, the array remains unchanged.
    ///
    /// - Parameter index: The index up to which the array should be truncated.
    ///
    /// Example Usage:
    /// ```
    /// var numbers = [1, 2, 3, 4, 5]
    /// numbers.truncate(to: 2)
    /// // numbers is now [1, 2, 3]
    /// ```
    mutating func truncate(to index: Int) {
        guard index < self.count else { return }
        self = Array(self[..<Swift.min(index + 1, self.count)])
    }
}
```